lib/DDG/Spice/CodeDay.pm

```diff
@@ -0,0 +1,56 @@
+package DDG::Spice::CodeDay;
+# ABSTRACT: Get all CodeDay events, sorted by closest to farthest
+
+use DDG::Spice;
+use Text::Trim;
+
+spice is_cached => 0;
+
+name "CodeDay";
+source "CodeDay.org";
+icon_url "https://codeday.org/assets/img/favicon.ico";
+description "Get information about a CodeDay or find the nearest one.";
+primary_example_queries "codeday near me", "when is codeday?", "codeday san diego";
+secondary_example_queries "codeday", "codeday seattle", "what is codeday bay area?";
+category "programming";
+topics "programming", "geek", "computing", "entertainment";
+code_url "https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/lib/DDG/Spice/CodeDay.pm";
+attribution github => ["tjhorner", "TJ Horner"],
+            twitter => "tjhorner";
+
+spice to => 'https://clear.codeday.org/api/regions/$1?term=$2&lat=$3&lng=$4&limit=50';
```

Comment:
@tjhorner this has now been deployed to beta, but it's not working properly. I suspect it's these calls API calls.

I think what we'll have to do, is always supply a lat long, by looking up the lat/long for the specified City. We have plans to expose a method for this in the future, but right now we don't have an easy way to do this.

I did some digging and came across http://damog.net/files/misc/zipcodes-csv-10-Aug-2004.zip which is a CSV containing the lat/lon for all US Zip Codes. This is overkill, but if we slim it down to the cities where there will be Code Day events, or maybe just the capitals for each state, we can easily load the file and lookup the lat long for it and make an API call.

How does that sound?